chapters/classes.tex

```diff
@@ -337,25 +337,26 @@ \subsection{Acyclic Bindings of Constants and Parameters}\label{acyclic-bindings
 
 \subsection{Component Variability Prefixes discrete, parameter, constant}\label{component-variability-prefixes-discrete-parameter-constant}
 
-The prefixes \lstinline!discrete!, \lstinline!parameter!, \lstinline!constant! of a component declaration are called \firstuse{variability prefixes}\index{variability!prefix}\index{declared variability}\index{variability!declared|see{declared variability}} and define in which situation the variable values of a component are initialized (see \cref{events-and-synchronization} and \cref{initialization-initial-equation-and-initial-algorithm}) and when they are changed in transient analysis (= solution of initial value problem of the hybrid DAE):
+The prefixes \lstinline!discrete!, \lstinline!parameter!, \lstinline!constant! of a component declaration are called \firstuse{variability prefixes}\index{variability!prefix}\index{component variability}\index{declared variability}\index{variability!declared|see{declared variability}} and define in which situation the variable values of a component are initialized (see \cref{events-and-synchronization} and \cref{initialization-initial-equation-and-initial-algorithm}) and when they are changed in transient analysis (= solution of initial value problem of the hybrid DAE):
 \begin{itemize}
 \item
   A variable \lstinline!vc! declared with \lstinline!constant!\indexinline{constant} prefix remains constant during transient analysis, with a value that is unaffected by the initialization problem.
+  This is called a \firstuse{constant}, or \firstuse{constant!variable}\index{constant!variable}\index{component variability!constant}.
 \item
   A variable \lstinline!vc! declared with the \lstinline!parameter!\indexinline{parameter} prefix remains constant during transient analysis, with a value determined by the initialization problem.
+  This is called a \firstuse{parameter}, or \firstuse{parameter!variable}\index{parameter!variable}\index{component variability!parameter}.
 \item
-  A \emph{discrete-time} variable \lstinline!vd! has a vanishing time derivative between events.
-  Note that this is not the same as saying that \lstinline!der(vd)=0! almost everywhere,
-  as the derivative is not even defined at the events, and it is not legal
-  to apply \lstinline!der! to discrete-time variables as they are not continuous. During transient analysis the variable
-  can only change its value at event
-  instants (see \cref{events-and-synchronization}).
+  A \emph{discrete-time}\index{discrete-time!variable}\index{component variability!discrete-time} variable \lstinline!vd! is a variable that is discrete-valued (that is, not of \lstinline!Real! type) or assigned in a \lstinline!when!-clause.
```

Comment:
What I was trying to suggest with `\firstuse` was simply this:
```suggestion
  A \firstuse{discrete-time variable}\index{discrete-time!variable}\index{component variability!discrete-time} \lstinline!vd! is a variable that is discrete-valued (that is, not of \lstinline!Real! type) or assigned in a \lstinline!when!-clause.
```

Reply:
Yes, but I read the comments and `firstuse` indicates that it later might automatically add the text to the index - and that will not work if the index should contain something slightly different as in this case. That's why I was hesitant to use it.

Reply:
I see.  However note that:
- There is currently one or `\index` calls with every `\firstuse`.
- The argument to `\index` is often, but not always the same as the argument to `\firstuse`.

My conclusion is that the comment is making more harm than good.  If we'd ever change `\firstuse` to also add to the index, we'd have to go through all the places anyway and remove the current `\index` calls that would then become redundant.

I see two possibilities for changing the comment and/or implementation of `\firstuse`:
- Maybe we should change the comment to settle that `\firstuse` is all about semantic markup for styling, and explain in the comment that it should go together with `\index`, and that it is sometimes the case that the argument to `\index` is structured a bit differently that the argument to `\firstuse`?
- Alternatively, we could give `\firstuse` an optional argument for how the introduced terminology should be presented in the index, and by default just use the mandatory argument as-is.

How about going with `\firstuse` instead of `\emph` in this case just to follow the same pattern as elsewhere, and open a separate issue about sorting out what `\firstuse` really should do and say it might do in the future?

Reply:
With the continuous-time variable item being fixed, this is now the only remaining unresolved comment.